source/solvers/postprocessing_cfd.cc

```diff
@@ -434,14 +435,30 @@ calculate_forces(
 {
   const FESystem<dim, dim> fe = dof_handler.get_fe();
 
-  double viscosity = physical_properties.fluids[0].viscosity;
+  double viscosity;
+  if (!physical_properties.non_newtonian_flow)
+    viscosity = physical_properties.fluids[0].viscosity;
+
+  // Non newtonian properties
+  std::unique_ptr<RheologicalModel<dim>> rheological_model;
+  if (physical_properties.non_newtonian_flow)
```

Comment:
Done! I generated a rheological_model.cc file since the methode model_cast() needs both Newtonian and Carreau classes to be instantiated before the definition of the method. I would really be much clearer if all methods from the rheological_model.h file were instantiated in the rheological_model.cc file (this should have been done in the first place, but I was very lazy I admit). A call to model_cast() should also replace the if conditions in the  GLSNavierStokesAssemblerNonNewtonianCore constructer in navier_stokes_assemblers.h.  Should I do it in this PR or in another one?